### PR TITLE
D3ASIM-1869: Fixed storage strategy self sufficiency calculation. Ada…

### PR DESCRIPTION
**Pull request by spyrostz**
…pted integration tests to also run for 2 sided pab market.

### CODE REVIEW

File: integration_tests/steps/integration_tests.py [hunk 1]
```diff
@@ -278,6 +278,16 @@ def run_sim_console(context, scenario, hours):
...
+@when('we run the d3a simulation on console with {scenario} for {hours} hrs '
+      'with default tick/slot length')
+def run_sim_console_decreased_tick_slot_length(context, scenario, hours):
+    context.export_path = os.path.join(context.simdir, scenario)
+    os.makedirs(context.export_path, exist_ok=True)
+    os.system("d3a -l FATAL run -d {hours}h -t 40s -s 40m --setup={scenario} "
+              "--export-path={export_path}"
+              .format(export_path=context.export_path, scenario=scenario, hours=hours))
+
+
```
Comment (hannesdiedrich) [inline integration_tests/steps/integration_tests.py:290]: As this is a copy of the code above these lines apart from the slot and tick length, I suggest to parse the slot and tick length in the step caption. 
Is the unusual slot and tick length needed here to test a corner case?

Comment (spyrostz) [reply integration_tests/steps/integration_tests.py:290]: Good idea, I'll do so, thanks. It is actually required in order for the 2-sided market offers-bids to match for the setup files that I used (essentially used the same setup files with the one sided market and adapted some parameters for 2 sided market to work exactly like the one sided). 40 is the max slot and tick length I could use actually. Any value lower than that works, but delays unnecessarily the simulation.

File: integration_tests/kpi.feature [hunk 2]
```diff
@@ -53,16 +53,24 @@ Feature: KPI integration tests
      And the penalties of the 'Penalty PV' is the sum of the residual available energy
      And the penalty cost of the 'Penalty PV' is respecting the penalty rate
 
-  Scenario: D3ASIM-2103, self sufficiency reports correct values on levels higher than houses
-     Given we have a scenario named kpi.d3asim_2103
+  Scenario Outline: D3ASIM-2103, self sufficiency reports correct values on levels higher than houses
+     Given we have a scenario named <scenario_file>
      And d3a is installed
-     When we run the d3a simulation on console with kpi.d3asim_2103 for 24 hrs
+     When we run the d3a simulation on console with <scenario_file> for 24 hrs (30, 30)
```
[the remark this answers is not in the document]
Comment (spyrostz) [reply integration_tests/kpi.feature:59]: Answered already here https://github.com/gridsingularity/d3a/pull/709#discussion_r403036674

File: src/d3a/d3a_core/sim_results/kpi.py [hunk 3]
```diff
@@ -76,7 +76,7 @@ def _accumulate_self_consumption_buffer(self, trade):
             self.self_consumption_buffer_wh += trade.offer.energy * 1000
 
     def _dissipate_self_consumption_buffer(self, trade):
-        if trade.seller_origin in self.ess_list:
+        if trade.seller_origin in self.ess_list and trade.seller_origin == trade.seller:
```
[the remark this answers is not in the document]
Comment (spyrostz) [reply src/d3a/d3a_core/sim_results/kpi.py:79]: Exactly

File: src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import os
```
[the remark this answers is not in the document]
Comment (spyrostz) [reply src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py:1]: I was thinking about that too, unfortunately this is not possible for 2 reasons:
1. Most important reason, the one-sided market setup file is not exactly the same as the 2 sided market one. In order to reproduce consistently the same trade pattern in both markets, I had to alter the pricing of the strategies too. Therefore there have to indeed be 2 different files. 
2. These tests are using the step that executes the simulation via the console. Therefore there is no easy way for altering the market type (of course, we can set env variables, that we then read in the setup file, but this approach seemed clunky to me)

File: src/d3a/setup/kpi/d3asim_2103_nopv1_2_sided_pab.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import os
```
[the remark this answers is not in the document]
Comment (spyrostz) [reply src/d3a/setup/kpi/d3asim_2103_nopv1_2_sided_pab.py:1]: I think this answer https://github.com/gridsingularity/d3a/pull/709#discussion_r403746866  applies here too.

File: integration_tests/kpi.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (faizan2590) [reply integration_tests/kpi.feature:59]: What I meant is since we have used 60secs as tick_length & 60mins as slot_length for almost all of our integration_tests, so I was wondering if you have observed any change in KPI results with a change in `tick_length & slot_length` ?

[the remark this answers is not in the document]
Comment (spyrostz) [reply integration_tests/kpi.feature:59]: Yes, with 60 seconds tick and slot length no trades take place. There are not enough ticks for the offer / bids to match

File: src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (faizan2590) [reply src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py:1]: Considering point 1, I would actually prefer to showcase the relatively change in trade thats is due to change in market that also lead to different KPI result. So others could could have a feeling of its affect due to change of market type with the same scenario.
Considering point 2, it could circumvented if we use this step `integration_tests/steps/integration_tests.py:611 --> we run the simulation with setup file {scenario} and parameters .. `

[the remark this answers is not in the document]
Comment (spyrostz) [reply src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py:1]: Considering point 1, I disagree, and I think there is a misunderstanding in what the test does on your side. The purpose of the test is not to check for different KPI results according to the market type. The purpose of the test is to check that, assuming that the 2 markets produce the same trades, the self sufficiency and self consumption members do not report different values.

[the remark this answers is not in the document]
Comment (faizan2590) [reply src/d3a/setup/kpi/d3asim_2103_2_sided_pab.py:1]: Thanks for explanation agreed now with point 1. Regarding point 2 its optional as I don't see any difference
